digdag-cli/src/main/java/io/digdag/cli/client/ShowTask.java

```diff
@@ -1,59 +1,314 @@
+    Format format = Format.TEXT;
+
+    @Parameter(names = {"-t", "--type"}, converter = TypeConverter.class)
+    Type type = Type.FULL;
+
     @Override
     public void mainWithClientException()
         throws Exception
     {
         if (args.size() != 1) {
             throw usage(null);
         }
-        showTasks(parseAttemptIdOrUsage(args.get(0)));
+        try {
+            show(parseAttemptIdOrUsage(args.get(0)));
+        }
+        catch (Throwable e) { e.printStackTrace(); }
     }
 
     public SystemExitException usage(String error)
     {
         err.println("Usage: " + programName + " tasks <attempt-id>");
         err.println("  Options:");
         showCommonOptions();
+        err.println("    -f, --format  FORMAT             Format: text or json (default: text)");
+        err.println("    -t, --type    TYPE               Information type: full or summary (default: full)");
         return systemExit(error);
     }
 
-    private void showTasks(Id attemptId)
+    private void show(Id attemptId)
         throws Exception
     {
         DigdagClient client = buildClient();
 
-        int count = 0;
-        for (RestTask task : client.getTasks(attemptId).getTasks()) {
-            ln("   id: %s", task.getId());
-            ln("   name: %s", task.getFullName());
-            ln("   state: %s", task.getState());
-            ln("   started: %s", task.getStartedAt().transform(TimeUtil::formatTime).or(""));
-            ln("   updated: %s", TimeUtil.formatTime(task.getUpdatedAt()));
-            ln("   config: %s", task.getConfig());
-            ln("   parent: %s", task.getParentId().orNull());
-            ln("   upstreams: %s", task.getUpstreams());
-            ln("   export params: %s", task.getExportParams());
-            ln("   store params: %s", task.getStoreParams());
-            ln("   state params: %s", task.getStateParams());
-            ln("");
-            count++;
-        }
-
-        if (count == 0) {
+        List<RestTask> tasks = client.getTasks(attemptId).getTasks();
+        if (tasks.size() == 0) {
             client.getSessionAttempt(attemptId);  // throws exception if attempt doesn't exist
         }
-        ln("%d entries.", count);
+
+        switch (type) {
+            case FULL:
+                format.printer.showTasks(this, tasks);
+                break;
+            case SUMMARY:
+                format.printer.showSummary(this, TasksSummary.fromTasks(tasks));
+                break;
+        }
+    }
+
+    static class TasksSummary
+    {
+        public final long totalTasks;
+        public final long totalInvokedTasks;
+        public final long totalSuccessTasks;
+
+        public final NullableLong averageStartDelayMillis;
+        public final NullableLong stdDevStartDelayMillis;
+
+        public final NullableLong averageExecTimeMillis;
+        public final NullableLong stdDevExecTimeMillis;
+
+        static class NullableLong
```

Comment:
Digdag database may still contain old archived tasks that don't have `started_at` and this class is to take care of the compatibility and to make TasksSummary simple.